pkg/server/io/container_io.go

```diff
@@ -86,13 +102,31 @@ func NewContainerIO(id string, opts ...ContainerIOOpts) (_ *ContainerIO, err err
 	if c.fifos == nil {
 		return nil, errors.New("fifos are not set")
 	}
-	// Create actual fifos.
-	stdio, closer, err := newStdioPipes(c.fifos)
+
+	u, err := url.Parse(c.fifos.Stdout)
+
 	if err != nil {
 		return nil, err
 	}
-	c.stdioPipes = stdio
-	c.closer = closer
+
+	if u.Scheme == "" {
+		u.Scheme = "fifo"
+	}
+
+	switch u.Scheme {
+	case "fifo":
+		// Create actual fifos.
+		stdio, closer, err := newStdioPipes(c.fifos)
+		if err != nil {
+			return nil, err
+		}
+		c.stdioPipes = stdio
+		c.closer = closer
+	case "binary":
+		stdio, closer := newEmptyPipes()
+		c.stdioPipes = stdio
+		c.closer = closer
```

Comment:
Think this can be:
```go
c.stdioPipes, c.closer = newEmptyPipes()
```

Reply:
yeah, true. no error here, so no potential state changes 👍